app/pages/api/[submit].js

```diff
@@ -0,0 +1,19 @@
+import { withSession } from 'next-session';
+import { postMiddleware } from '../../form-schema';
+
+function runMiddleware(req, res, fn) {
+  return new Promise((resolve, reject) => {
+    fn(req, res, (result) => {
+      if (result instanceof Error) {
+        return reject(result);
+      }
+      return resolve(result);
+    });
+  });
+}
```

Comment:
This is just taken straight from the [nextjs docs](https://nextjs.org/docs/api-routes/api-middlewares#connectexpress-middleware-support), its a wrapper to allow running middleware on post requests